client/beefy/src/metrics.rs

```diff
@@ -18,16 +18,22 @@
 
 //! BEEFY Prometheus metrics definition
 
+use log::debug;
 use prometheus::{register, Counter, Gauge, PrometheusError, Registry, U64};
 
-/// BEEFY metrics exposed through Prometheus
-pub(crate) struct Metrics {
+/// Helper trait for registering BEEFY metrics to Prometheus registry.
+pub(crate) trait PrometheusRegister<T: Sized = Self>: Sized {
+	const DESCRIPTION: &'static str;
+	fn register(registry: &Registry) -> Result<Self, PrometheusError>;
+}
+
+/// BEEFY voting-related metrics exposed through Prometheus
+#[derive(Clone, Debug)]
+pub struct VoterMetrics {
 	/// Current active validator set id
 	pub beefy_validator_set_id: Gauge<U64>,
 	/// Total number of votes sent by this node
 	pub beefy_votes_sent: Counter<U64>,
-	/// Most recent concluded voting round
-	pub beefy_round_concluded: Gauge<U64>,
```

Comment:
naive q: not obvious to me what this metric is replaced by - what's it taken out for beyond not strictly increasing?

Reply:
Found this metric to be redundant. It'll always have same value as `best_beefy_block`